Multi-project @QuarkusTest in IDE to use compiled source

**Describe the bug**
Running @QuarkusTests in Eclipse IDE for a multi-project Maven setup, the tests are picking up the jar dependencies not the compiled source.

**Environment (please complete the following information):**
 - Quarkus version or git rev: 1.3.0.Alpha1

**Additional context**
https://groups.google.com/d/msg/quarkus-dev/ovaJ1GUang8/9jUbs9m7AQAJ

Commit looks good.

It is deliberate that the parent pom does not have the aggregator pom as its parent.  This is a pretty common practice.

I think if you delete any pre-built jars from your target directories / repo you will then get an error like this where it appears to be looking for the jars:

> java.lang.RuntimeException: io.quarkus.bootstrap.BootstrapException: Failed to create the application model for org.acme:quarkus-quickstart-multimodule-main::jar:1.0-SNAPSHOT
	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:129)
	at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:179)
	at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:207)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$7(ClassBasedTestDescriptor.java:359)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:359)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:189)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:78)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:132)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:229)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:197)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:211)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:191)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:137)
	at org.eclipse.jdt.internal.junit5.runner.JUnit5TestReference.run(JUnit5TestReference.java:89)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:41)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:542)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:770)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:464)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210)
Caused by: io.quarkus.bootstrap.BootstrapException: Failed to create the application model for org.acme:quarkus-quickstart-multimodule-main::jar:1.0-SNAPSHOT
	at io.quarkus.bootstrap.BootstrapAppModelFactory.resolveAppModel(BootstrapAppModelFactory.java:293)
	at io.quarkus.bootstrap.app.QuarkusBootstrap.bootstrap(QuarkusBootstrap.java:134)
	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:89)
	... 39 more
Caused by: io.quarkus.bootstrap.resolver.AppModelResolverException: Failed to resolve dependencies for org.acme:quarkus-quickstart-multimodule-main:jar:1.0-SNAPSHOT
	at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.resolveManagedDependencies(MavenArtifactResolver.java:276)
	at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.doResolveModel(BootstrapAppModelResolver.java:174)
	at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.resolveManagedModel(BootstrapAppModelResolver.java:149)
	at io.quarkus.bootstrap.BootstrapAppModelFactory.resolveAppModel(BootstrapAppModelFactory.java:279)
	... 41 more
Caused by: org.eclipse.aether.resolution.DependencyResolutionException: The following artifacts could not be resolved: org.acme:quarkus-quickstart-multimodule-html:jar:1.0-SNAPSHOT, org.acme:quarkus-quickstart-multimodule-rest:jar:1.0-SNAPSHOT: Could not find artifact org.acme:quarkus-quickstart-multimodule-html:jar:1.0-SNAPSHOT
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:355)
	at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.resolveManagedDependencies(MavenArtifactResolver.java:272)
	... 44 more
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: The following artifacts could not be resolved: org.acme:quarkus-quickstart-multimodule-html:jar:1.0-SNAPSHOT, org.acme:quarkus-quickstart-multimodule-rest:jar:1.0-SNAPSHOT: Could not find artifact org.acme:quarkus-quickstart-multimodule-html:jar:1.0-SNAPSHOT
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:422)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:224)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:338)
	... 45 more
Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.acme:quarkus-quickstart-multimodule-html:jar:1.0-SNAPSHOT
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:412)
	... 47 more

I see that in your case once you do have the jars you can modify the source, but I'm guessing that maybe you're getting an extra hot-deployment action with that web resource that I'm not seeing with my pojo dependencies?

With your original combined aggregator/parent pom the jars did not need to exist in order to run the test.

> @geoand would you have a time to check out https://github.com/aloubyansky/playground/tree/quarkus-test-multimodule, try changing HelloResource in the IDE and running HelloResourceTest w/o installing/building the project from the command line?

I tried what you proposwd and here is the output:

```
/home/gandrian/.sdkman/candidates/java/11.0.6-zulu/bin/java -ea -Didea.test.cyclic.buffer.size=1048576 -javaagent:/home/gandrian/software/ideaIU-193.3519.25/idea-IU-193.3519.25/lib/idea_rt.jar=39989:/home/gandrian/software/ideaIU-193.3519.25/idea-IU-193.3519.25/bin -Dfile.encoding=UTF-8 -classpath /home/gandrian/software/ideaIU-193.3519.25/idea-IU-193.3519.25/lib/idea_rt.jar:/home/gandrian/software/ideaIU-193.3519.25/idea-IU-193.3519.25/plugins/junit/lib/junit5-rt.jar:/home/gandrian/software/ideaIU-193.3519.25/idea-IU-193.3519.25/plugins/junit/lib/junit-rt.jar:/home/gandrian/Desktop/playground/runner/target/test-classes:/home/gandrian/Desktop/playground/runner/target/classes:/home/gandrian/.m2/repository/io/quarkus/quarkus-resteasy/1.3.0.Alpha1/quarkus-resteasy-1.3.0.Alpha1.jar:/home/gandrian/.m2/repository/io/quarkus/quarkus-vertx-http/1.3.0.Alpha1/quarkus-vertx-http-1.3.0.Alpha1.jar:/home/gandrian/.m2/repository/io/quarkus/quarkus-development-mode-spi/1.3.0.Alpha1/quarkus-development-mode-spi-1.3.0.Alpha1.jar:/home/gandrian/.m2/repository/io/quarkus/security/quarkus-security/1.0.1.Final/quarkus-security-1.0.1.Final.jar:/home/gandrian/.m2/repository/jakarta/enterprise/jakarta.enterprise.cdi-api/2.0.2/jakarta.enterprise.cdi-api-2.0.2.jar:/home/gandrian/.m2/repository/jakarta/el/jakarta.el-api/3.0.3/jakarta.el-api-3.0.3.jar:/home/gandrian/.m2/repository/jakarta/interceptor/jakarta.interceptor-api/1.2.5/jakarta.interceptor-api-1.2.5.jar:/home/gandrian/.m2/repository/io/quarkus/quarkus-vertx-core/1.3.0.Alpha1/quarkus-vertx-core-1.3.0.Alpha1.jar:/home/gandrian/.m2/repository/io/quarkus/quarkus-netty/1.3.0.Alpha1/quarkus-netty-1.3.0.Alpha1.jar:/home/gandrian/.m2/repository/io/netty/netty-codec/4.1.44.Final/netty-codec-4.1.44.Final.jar:/home/gandrian/.m2/repository/io/netty/netty-handler/4.1.44.Final/netty-handler-4.1.44.Final.jar:/home/gandrian/.m2/repository/io/vertx/vertx-core/3.8.5/vertx-core-3.8.5.jar:/home/gandrian/.m2/repository/io/netty/netty-common/4.1.44.Final/netty-common-4.1.44.Final.jar:/home/gandrian/.m2/repository/io/netty/netty-buffer/4.1.44.Final/netty-buffer-4.1.44.Final.jar:/home/gandrian/.m2/repository/io/netty/netty-transport/4.1.44.Final/netty-transport-4.1.44.Final.jar:/home/gandrian/.m2/repository/io/netty/netty-handler-proxy/4.1.44.Final/netty-handler-proxy-4.1.44.Final.jar:/home/gandrian/.m2/repository/io/netty/netty-codec-socks/4.1.44.Final/netty-codec-socks-4.1.44.Final.jar:/home/gandrian/.m2/repository/io/netty/netty-codec-http/4.1.44.Final/netty-codec-http-4.1.44.Final.jar:/home/gandrian/.m2/repository/io/netty/netty-codec-http2/4.1.44.Final/netty-codec-http2-4.1.44.Final.jar:/home/gandrian/.m2/repository/io/netty/netty-resolver/4.1.44.Final/netty-resolver-4.1.44.Final.jar:/home/gandrian/.m2/repository/io/netty/netty-resolver-dns/4.1.44.Final/netty-resolver-dns-4.1.44.Final.jar:/home/gandrian/.m2/repository/io/netty/netty-codec-dns/4.1.44.Final/netty-codec-dns-4.1.44.Final.jar:/home/gandrian/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.10.2/jackson-core-2.10.2.jar:/home/gandrian/.m2/repository/io/vertx/vertx-web/3.8.5/vertx-web-3.8.5.jar:/home/gandrian/.m2/repository/io/vertx/vertx-web-common/3.8.5/vertx-web-common-3.8.5.jar:/home/gandrian/.m2/repository/io/vertx/vertx-auth-common/3.8.5/vertx-auth-common-3.8.5.jar:/home/gandrian/.m2/repository/io/vertx/vertx-bridge-common/3.8.5/vertx-bridge-common-3.8.5.jar:/home/gandrian/.m2/repository/io/quarkus/quarkus-resteasy-server-common/1.3.0.Alpha1/quarkus-resteasy-server-common-1.3.0.Alpha1.jar:/home/gandrian/.m2/repository/io/quarkus/quarkus-arc/1.3.0.Alpha1/quarkus-arc-1.3.0.Alpha1.jar:/home/gandrian/.m2/repository/io/quarkus/arc/arc/1.3.0.Alpha1/arc-1.3.0.Alpha1.jar:/home/gandrian/.m2/repository/jakarta/transaction/jakarta.transaction-api/1.3.3/jakarta.transaction-api-1.3.3.jar:/home/gandrian/.m2/repository/org/eclipse/microprofile/context-propagation/microprofile-context-propagation-api/1.0.1/microprofile-context-propagation-api-1.0.1.jar:/home/gandrian/.m2/repository/io/quarkus/quarkus-resteasy-common/1.3.0.Alpha1/quarkus-resteasy-common-1.3.0.Alpha1.jar:/home/gandrian/.m2/repository/org/jboss/resteasy/resteasy-core/4.4.2.Final/resteasy-core-4.4.2.Final.jar:/home/gandrian/.m2/repository/org/jboss/spec/javax/ws/rs/jboss-jaxrs-api_2.1_spec/2.0.1.Final/jboss-jaxrs-api_2.1_spec-2.0.1.Final.jar:/home/gandrian/.m2/repository/org/jboss/spec/javax/xml/bind/jboss-jaxb-api_2.3_spec/2.0.0.Final/jboss-jaxb-api_2.3_spec-2.0.0.Final.jar:/home/gandrian/.m2/repository/org/jboss/resteasy/resteasy-core-spi/4.4.2.Final/resteasy-core-spi-4.4.2.Final.jar:/home/gandrian/.m2/repository/org/reactivestreams/reactive-streams/1.0.3/reactive-streams-1.0.3.jar:/home/gandrian/.m2/repository/org/eclipse/microprofile/config/microprofile-config-api/1.3/microprofile-config-api-1.3.jar:/home/gandrian/.m2/repository/com/sun/activation/jakarta.activation/1.2.1/jakarta.activation-1.2.1.jar:/home/gandrian/.m2/repository/jakarta/validation/jakarta.validation-api/2.0.2/jakarta.validation-api-2.0.2.jar:/home/gandrian/Desktop/playground/html/target/classes:/home/gandrian/Desktop/playground/rest/target/classes:/home/gandrian/.m2/repository/io/quarkus/quarkus-junit5/1.3.0.Alpha1/quarkus-junit5-1.3.0.Alpha1.jar:/home/gandrian/.m2/repository/io/quarkus/quarkus-bootstrap-core/1.3.0.Alpha1/quarkus-bootstrap-core-1.3.0.Alpha1.jar:/home/gandrian/.m2/repository/org/ow2/asm/asm/7.3.1/asm-7.3.1.jar:/home/gandrian/.m2/repository/org/apache/maven/maven-embedder/3.5.4/maven-embedder-3.5.4.jar:/home/gandrian/.m2/repository/org/apache/maven/maven-settings/3.5.4/maven-settings-3.5.4.jar:/home/gandrian/.m2/repository/org/apache/maven/maven-core/3.5.4/maven-core-3.5.4.jar:/home/gandrian/.m2/repository/org/apache/maven/maven-artifact/3.5.4/maven-artifact-3.5.4.jar:/home/gandrian/.m2/repository/org/apache/maven/maven-plugin-api/3.5.4/maven-plugin-api-3.5.4.jar:/home/gandrian/.m2/repository/org/apache/maven/maven-model/3.5.4/maven-model-3.5.4.jar:/home/gandrian/.m2/repository/org/apache/maven/maven-model-builder/3.5.4/maven-model-builder-3.5.4.jar:/home/gandrian/.m2/repository/org/apache/maven/maven-builder-support/3.5.4/maven-builder-support-3.5.4.jar:/home/gandrian/.m2/repository/org/apache/maven/resolver/maven-resolver-api/1.1.1/maven-resolver-api-1.1.1.jar:/home/gandrian/.m2/repository/org/apache/maven/resolver/maven-resolver-util/1.1.1/maven-resolver-util-1.1.1.jar:/home/gandrian/.m2/repository/org/apache/maven/shared/maven-shared-utils/3.2.1/maven-shared-utils-3.2.1.jar:/home/gandrian/.m2/repository/commons-io/commons-io/2.6/commons-io-2.6.jar:/home/gandrian/.m2/repository/com/google/inject/guice/4.2.0/guice-4.2.0-no_aop.jar:/home/gandrian/.m2/repository/aopalliance/aopalliance/1.0/aopalliance-1.0.jar:/home/gandrian/.m2/repository/com/google/guava/guava/27.0.1-jre/guava-27.0.1-jre.jar:/home/gandrian/.m2/repository/com/google/guava/failureaccess/1.0.1/failureaccess-1.0.1.jar:/home/gandrian/.m2/repository/com/google/guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar:/home/gandrian/.m2/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar:/home/gandrian/.m2/repository/org/checkerframework/checker-qual/2.5.2/checker-qual-2.5.2.jar:/home/gandrian/.m2/repository/com/google/errorprone/error_prone_annotations/2.2.0/error_prone_annotations-2.2.0.jar:/home/gandrian/.m2/repository/com/google/j2objc/j2objc-annotations/1.1/j2objc-annotations-1.1.jar:/home/gandrian/.m2/repository/org/codehaus/mojo/animal-sniffer-annotations/1.17/animal-sniffer-annotations-1.17.jar:/home/gandrian/.m2/repository/org/codehaus/plexus/plexus-utils/3.0.24/plexus-utils-3.0.24.jar:/home/gandrian/.m2/repository/org/codehaus/plexus/plexus-classworlds/2.5.2/plexus-classworlds-2.5.2.jar:/home/gandrian/.m2/repository/org/eclipse/sisu/org.eclipse.sisu.plexus/0.3.3/org.eclipse.sisu.plexus-0.3.3.jar:/home/gandrian/.m2/repository/org/codehaus/plexus/plexus-component-annotations/1.7.1/plexus-component-annotations-1.7.1.jar:/home/gandrian/.m2/repository/commons-cli/commons-cli/1.4/commons-cli-1.4.jar:/home/gandrian/.m2/repository/org/apache/maven/maven-settings-builder/3.5.4/maven-settings-builder-3.5.4.jar:/home/gandrian/.m2/repository/org/codehaus/plexus/plexus-interpolation/1.24/plexus-interpolation-1.24.jar:/home/gandrian/.m2/repository/org/sonatype/plexus/plexus-sec-dispatcher/1.4/plexus-sec-dispatcher-1.4.jar:/home/gandrian/.m2/repository/org/sonatype/plexus/plexus-cipher/1.4/plexus-cipher-1.4.jar:/home/gandrian/.m2/repository/org/apache/maven/maven-resolver-provider/3.5.4/maven-resolver-provider-3.5.4.jar:/home/gandrian/.m2/repository/org/apache/maven/maven-repository-metadata/3.5.4/maven-repository-metadata-3.5.4.jar:/home/gandrian/.m2/repository/org/apache/maven/resolver/maven-resolver-spi/1.1.1/maven-resolver-spi-1.1.1.jar:/home/gandrian/.m2/repository/org/apache/maven/resolver/maven-resolver-impl/1.1.1/maven-resolver-impl-1.1.1.jar:/home/gandrian/.m2/repository/org/apache/maven/resolver/maven-resolver-connector-basic/1.1.1/maven-resolver-connector-basic-1.1.1.jar:/home/gandrian/.m2/repository/org/apache/maven/resolver/maven-resolver-transport-wagon/1.1.1/maven-resolver-transport-wagon-1.1.1.jar:/home/gandrian/.m2/repository/org/apache/maven/wagon/wagon-provider-api/3.0.0/wagon-provider-api-3.0.0.jar:/home/gandrian/.m2/repository/org/apache/maven/wagon/wagon-http/3.0.0/wagon-http-3.0.0.jar:/home/gandrian/.m2/repository/org/apache/maven/wagon/wagon-http-shared/3.0.0/wagon-http-shared-3.0.0.jar:/home/gandrian/.m2/repository/org/jsoup/jsoup/1.7.2/jsoup-1.7.2.jar:/home/gandrian/.m2/repository/org/apache/maven/wagon/wagon-file/3.0.0/wagon-file-3.0.0.jar:/home/gandrian/.m2/repository/org/jboss/logging/jboss-logging/3.3.2.Final/jboss-logging-3.3.2.Final.jar:/home/gandrian/.m2/repository/org/jboss/logging/commons-logging-jboss-logging/1.0.0.Final/commons-logging-jboss-logging-1.0.0.Final.jar:/home/gandrian/.m2/repository/io/quarkus/quarkus-test-common/1.3.0.Alpha1/quarkus-test-common-1.3.0.Alpha1.jar:/home/gandrian/.m2/repository/io/quarkus/quarkus-core-deployment/1.3.0.Alpha1/quarkus-core-deployment-1.3.0.Alpha1.jar:/home/gandrian/.m2/repository/io/quarkus/gizmo/gizmo/1.0.2.Final/gizmo-1.0.2.Final.jar:/home/gandrian/.m2/repository/org/ow2/asm/asm-util/7.3.1/asm-util-7.3.1.jar:/home/gandrian/.m2/repository/org/ow2/asm/asm-tree/7.3.1/asm-tree-7.3.1.jar:/home/gandrian/.m2/repository/org/ow2/asm/asm-analysis/7.3.1/asm-analysis-7.3.1.jar:/home/gandrian/.m2/repository/io/quarkus/quarkus-builder/1.3.0.Alpha1/quarkus-builder-1.3.0.Alpha1.jar:/home/gandrian/.m2/repository/io/quarkus/quarkus-jsonp-deployment/1.3.0.Alpha1/quarkus-jsonp-deployment-1.3.0.Alpha1.jar:/home/gandrian/.m2/repository/io/quarkus/quarkus-jsonp/1.3.0.Alpha1/quarkus-jsonp-1.3.0.Alpha1.jar:/home/gandrian/.m2/repository/org/glassfish/jakarta.json/1.1.6/jakarta.json-1.1.6.jar:/home/gandrian/.m2/repository/org/jboss/jandex/2.1.2.Final/jandex-2.1.2.Final.jar:/home/gandrian/.m2/repository/org/junit/jupiter/junit-jupiter/5.6.0/junit-jupiter-5.6.0.jar:/home/gandrian/.m2/repository/org/junit/jupiter/junit-jupiter-api/5.6.0/junit-jupiter-api-5.6.0.jar:/home/gandrian/.m2/repository/org/apiguardian/apiguardian-api/1.1.0/apiguardian-api-1.1.0.jar:/home/gandrian/.m2/repository/org/opentest4j/opentest4j/1.2.0/opentest4j-1.2.0.jar:/home/gandrian/.m2/repository/org/junit/platform/junit-platform-commons/1.6.0/junit-platform-commons-1.6.0.jar:/home/gandrian/.m2/repository/org/junit/jupiter/junit-jupiter-params/5.6.0/junit-jupiter-params-5.6.0.jar:/home/gandrian/.m2/repository/org/junit/jupiter/junit-jupiter-engine/5.6.0/junit-jupiter-engine-5.6.0.jar:/home/gandrian/.m2/repository/org/junit/platform/junit-platform-engine/1.6.0/junit-platform-engine-1.6.0.jar:/home/gandrian/.m2/repository/io/quarkus/quarkus-core/1.3.0.Alpha1/quarkus-core-1.3.0.Alpha1.jar:/home/gandrian/.m2/repository/jakarta/annotation/jakarta.annotation-api/1.3.5/jakarta.annotation-api-1.3.5.jar:/home/gandrian/.m2/repository/jakarta/inject/jakarta.inject-api/1.0/jakarta.inject-api-1.0.jar:/home/gandrian/.m2/repository/io/smallrye/config/smallrye-config/1.5.1/smallrye-config-1.5.1.jar:/home/gandrian/.m2/repository/io/smallrye/config/smallrye-config-common/1.5.1/smallrye-config-common-1.5.1.jar:/home/gandrian/.m2/repository/org/jboss/logmanager/jboss-logmanager-embedded/1.0.4/jboss-logmanager-embedded-1.0.4.jar:/home/gandrian/.m2/repository/org/jboss/logging/jboss-logging-annotations/2.1.0.Final/jboss-logging-annotations-2.1.0.Final.jar:/home/gandrian/.m2/repository/org/jboss/threads/jboss-threads/3.0.1.Final/jboss-threads-3.0.1.Final.jar:/home/gandrian/.m2/repository/org/slf4j/slf4j-api/1.7.29/slf4j-api-1.7.29.jar:/home/gandrian/.m2/repository/org/jboss/slf4j/slf4j-jboss-logging/1.2.0.Final/slf4j-jboss-logging-1.2.0.Final.jar:/home/gandrian/.m2/repository/org/graalvm/sdk/graal-sdk/19.3.1/graal-sdk-19.3.1.jar:/home/gandrian/.m2/repository/org/wildfly/common/wildfly-common/1.5.3.Final-format-001/wildfly-common-1.5.3.Final-format-001.jar:/home/gandrian/.m2/repository/io/rest-assured/rest-assured/4.2.0/rest-assured-4.2.0.jar:/home/gandrian/.m2/repository/org/codehaus/groovy/groovy/2.5.8/groovy-2.5.8.jar:/home/gandrian/.m2/repository/org/codehaus/groovy/groovy-xml/2.5.8/groovy-xml-2.5.8.jar:/home/gandrian/.m2/repository/org/apache/httpcomponents/httpclient/4.5.11/httpclient-4.5.11.jar:/home/gandrian/.m2/repository/org/apache/httpcomponents/httpcore/4.4.13/httpcore-4.4.13.jar:/home/gandrian/.m2/repository/commons-codec/commons-codec/1.13/commons-codec-1.13.jar:/home/gandrian/.m2/repository/org/apache/httpcomponents/httpmime/4.5.3/httpmime-4.5.3.jar:/home/gandrian/.m2/repository/org/hamcrest/hamcrest/2.1/hamcrest-2.1.jar:/home/gandrian/.m2/repository/org/ccil/cowan/tagsoup/tagsoup/1.2.1/tagsoup-1.2.1.jar:/home/gandrian/.m2/repository/io/rest-assured/json-path/4.2.0/json-path-4.2.0.jar:/home/gandrian/.m2/repository/org/codehaus/groovy/groovy-json/2.5.8/groovy-json-2.5.8.jar:/home/gandrian/.m2/repository/io/rest-assured/rest-assured-common/4.2.0/rest-assured-common-4.2.0.jar:/home/gandrian/.m2/repository/io/rest-assured/xml-path/4.2.0/xml-path-4.2.0.jar:/home/gandrian/.m2/repository/org/apache/commons/commons-lang3/3.9/commons-lang3-3.9.jar:/home/gandrian/.m2/repository/jakarta/xml/bind/jakarta.xml.bind-api/2.3.2/jakarta.xml.bind-api-2.3.2.jar:/home/gandrian/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.1/jakarta.activation-api-1.2.1.jar:/home/gandrian/.m2/repository/org/apache/sling/org.apache.sling.javax.activation/0.1.0/org.apache.sling.javax.activation-0.1.0.jar:/home/gandrian/.m2/repository/org/junit/platform/junit-platform-launcher/1.6.0/junit-platform-launcher-1.6.0.jar com.intellij.rt.junit.JUnitStarter -ideVersion5 -junit5 org.acme.quarkus.sample.HelloResourceTest
Feb 15, 2020 12:44:53 PM org.jboss.threads.Version <clinit>
INFO: JBoss Threads version 3.0.1.Final
Feb 15, 2020 12:44:54 PM io.quarkus.runtime.Timing printStartupTime
INFO: Quarkus 1.3.0.Alpha1 started in 1.514s. 
Feb 15, 2020 12:44:54 PM io.quarkus.runtime.Timing printStartupTime
INFO: Profile test activated. 
Feb 15, 2020 12:44:54 PM io.quarkus.runtime.Timing printStartupTime
INFO: Installed features: []


Feb 15, 2020 12:44:54 PM io.quarkus.runtime.Timing printStopTime
INFO: Quarkus stopped in 0.000s

org.junit.jupiter.api.extension.TestInstantiationException: Failed to create test instance

	at io.quarkus.test.junit.QuarkusTestExtension.initTestState(QuarkusTestExtension.java:280)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:264)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.api.extension.InvocationInterceptor.interceptTestClassConstructor(InvocationInterceptor.java:72)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:77)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeTestClassConstructor(ClassBasedTestDescriptor.java:333)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.instantiateTestClass(ClassBasedTestDescriptor.java:280)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.instantiateTestClass(ClassTestDescriptor.java:77)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.instantiateAndPostProcessTestInstance(ClassBasedTestDescriptor.java:262)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$testInstancesProvider$2(ClassBasedTestDescriptor.java:256)
	at java.base/java.util.Optional.orElseGet(Optional.java:369)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$testInstancesProvider$3(ClassBasedTestDescriptor.java:255)
	at org.junit.jupiter.engine.execution.TestInstancesProvider.getTestInstances(TestInstancesProvider.java:29)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$prepare$0(TestMethodTestDescriptor.java:108)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.prepare(TestMethodTestDescriptor.java:107)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.prepare(TestMethodTestDescriptor.java:71)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$prepare$1(NodeTestTask.java:107)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.prepare(NodeTestTask.java:107)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:75)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:248)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$5(DefaultLauncher.java:211)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:226)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:199)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:132)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at io.quarkus.runner.bootstrap.RunningQuarkusApplicationImpl.instance(RunningQuarkusApplicationImpl.java:80)
	at io.quarkus.test.junit.QuarkusTestExtension.initTestState(QuarkusTestExtension.java:273)
	... 59 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.runner.bootstrap.RunningQuarkusApplicationImpl.instance(RunningQuarkusApplicationImpl.java:75)
	... 60 more
Caused by: java.lang.IllegalStateException: Unable to locate CDIProvider
	at javax.enterprise.inject.spi.CDI.findAllProviders(CDI.java:121)
	at javax.enterprise.inject.spi.CDI.getCDIProvider(CDI.java:82)
	at javax.enterprise.inject.spi.CDI.current(CDI.java:64)
	... 65 more



Process finished with exit code 255

```

--- a/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/LocalProject.java
+++ b/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/LocalProject.java
@@ -80,25 +80,20 @@ public class LocalProject {
 
     private static Model loadRootModel(Path currentProjectDir) throws BootstrapException {
         Path pomXml = currentProjectDir.resolve(POM_XML);
-        Model model = readModel(pomXml);
-        Parent parent = model.getParent();
-        while (parent != null) {
-            if (parent.getRelativePath() != null && !parent.getRelativePath().isEmpty()) {
+        Model model = null;
+        while (Files.exists(pomXml)) {
+            model = readModel(pomXml);
+            final Parent parent = model.getParent();
+            if (parent != null
+                    && parent.getRelativePath() != null
+                    && !parent.getRelativePath().isEmpty()) {
                 pomXml = pomXml.getParent().resolve(parent.getRelativePath()).normalize();
-                if (!Files.exists(pomXml)) {
-                    return model;
-                }
                 if (Files.isDirectory(pomXml)) {
                     pomXml = pomXml.resolve(POM_XML);
                 }
             } else {
                 pomXml = pomXml.getParent().getParent().resolve(POM_XML);
-                if (!Files.exists(pomXml)) {
-                    return model;
-                }
             }
-            model = readModel(pomXml);
-            parent = model.getParent();
         }
         return model;
     }

--- a/independent-projects/bootstrap/core/src/test/java/io/quarkus/bootstrap/workspace/test/LocalWorkspaceDiscoveryTest.java
+++ b/independent-projects/bootstrap/core/src/test/java/io/quarkus/bootstrap/workspace/test/LocalWorkspaceDiscoveryTest.java
@@ -107,10 +107,10 @@ public class LocalWorkspaceDiscoveryTest {
         assertEquals("independent", project.getArtifactId());
         assertEquals(MvnProjectBuilder.DEFAULT_VERSION, project.getVersion());
         final Map<AppArtifactKey, LocalProject> projects = project.getWorkspace().getProjects();
-        assertEquals(1, projects.size());
+        assertEquals(6, projects.size());
         assertTrue(projects.containsKey(new AppArtifactKey(MvnProjectBuilder.DEFAULT_GROUP_ID, "independent")));
 
-        assertLocalDeps(project);
+        assertLocalDeps(project, "root-module-not-direct-child", "root-no-parent-module", "root-module-with-parent");
     }
 
     @Test
@@ -135,9 +135,9 @@ public class LocalWorkspaceDiscoveryTest {
         assertEquals(MvnProjectBuilder.DEFAULT_VERSION, project.getVersion());
         assertNotNull(project.getWorkspace());
         final Map<AppArtifactKey, LocalProject> projects = project.getWorkspace().getProjects();
-        assertEquals(1, projects.size());
+        assertEquals(5, projects.size());
         assertTrue(projects.containsKey(new AppArtifactKey(MvnProjectBuilder.DEFAULT_GROUP_ID, "root-no-parent-module")));
-        assertLocalDeps(project);
+        assertLocalDeps(project, "root-module-not-direct-child");
     }
 
     @Test

